.github/workflows/validate.yml

```diff
@@ -11,58 +11,54 @@ jobs:
     strategy:
       matrix:
         BUILD_TYPE: [Release, Debug]
+        DISTRO: ['fedora', 'fedora_rawhide', 'fedora_mpich', 'opensuse', 'ubuntu', 'ubuntu_devel', 'ubuntu_rolling', 'ubuntu_mpich']
+        CXX: ['g++', 'clang++']
+        EXTERNAL_MPI4PY: ['On','Off']
+        include:
+          - BUILD_TYPE: 'Debug'
+            DISTRO: 'fedora_intel'
+            CXX: 'icpc'
+            EXTERNAL_MPI4PY: 'On'
+          - BUILD_TYPE: 'Debug'
+            DISTRO: 'fedora_intel-oneapi'
+            CXX: 'icpx'
+            EXTERNAL_MPI4PY: 'On'
 
     runs-on: ubuntu-latest
+    container: espressopp/buildenv:${{ matrix.distro }}
     steps:
+      - uses: actions/cache@v2
+        with:
+          path: ~/.ccache
+          key: ${{ matrix.DISTRO }}-${{ matrix.CXX }}-${{ matrix.BUILD_TYPE }}-${{ github.run_id }}
+          restore-keys: ${{ matrix.DISTRO }}-${{ matrix.CXX }}-${{ matrix.BUILD_TYPE }}
+
       - uses: actions/checkout@v2
 
-      - name: Dependencies
+      - name: Get trail license
+        if: ${{ matrix.CXX == 'icpc' }}
         run: |
-          sudo apt-get update
-          sudo apt-get install -yq \
-          libboost-all-dev \
-          libfftw3-dev \
-          openmpi-bin \
-          python3-dev \
-          python3-h5py
-
-
-
+          mkdir ~/Licenses
+          curl https://dynamicinstaller.intel.com/api/v2/license > ~/Licenses/intel.lic
+      
       - name: Configure
-        run: cmake -S ${{github.workspace}} -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{ matrix.BUILD_TYPE }} -DEXTERNAL_MPI4PY=OFF
+        run: cmake -S . -B build -DCMAKE_BUILD_TYPE=${{ matrix.BUILD_TYPE }} -DCMAKE_CXX_COMPILER=${{ matrix.CXX }} -DEXTERNAL_MPI4PY=${{ matrix.EXTERNAL_MPI4PY }}
```

Comment:
We will soon also need the `hdf5` library which is a C library. Please also add the C compilers here.

Reply:
We can do that when it actually happened. Plus you could write the implement as `extern C`, which would still only require a c++ compiler.

Honestly I would prefer only having one compiler if possible as compiler mixing is another source of errors that can confuse users.

Reply:
https://github.com/votca/csg/blob/master/src/libcsg/modules/io/h5mdtrajectoryreader.cc

Reply:
Well, you are only looking for the `CXX` components.
https://github.com/votca/csg/blob/master/src/libcsg/CMakeLists.txt#L18
Interesting this is working since you are using the `C` interface.

If you look for `C` and `HL`, you get the following error:
```
  Unknown extension ".c" for file

   .../espressopp/cmake-build-debug/CMakeFiles/hdf5/cmake_hdf5_test.c

  try_compile() works only for enabled languages.  Currently these are:

    CXX

  See project() command to enable other languages.
```
Can be resolved by adding `C` as a language, however, this requires `C` compilers.

Reply:
> We should remove the pull request trigger. This is already covered by the push trigger and enforced by the branch protection settings.

Nope it not the same the pull request trigger tests the commit merged with the target branch, while the push trigger just tests the pushed commit itself. I think the former is more important.

Reply:
I see, we will have to restructure the matix build then.